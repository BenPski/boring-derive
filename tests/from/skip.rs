use boring_derive::From;

#[derive(Debug, From)]
enum Example {
    Nothing,
    #[from(skip)]
    Number(f32),
    Str(String),
}

fn main() {
    let ex: Example = 1.3.into();
    println!("{:?}", ex);
}
