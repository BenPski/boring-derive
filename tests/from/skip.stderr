error[E0277]: the trait bound `Example: From<{float}>` is not satisfied
  --> tests/from/skip.rs:12:27
   |
12 |     let ex: Example = 1.3.into();
   |                           ^^^^ the trait `From<{float}>` is not implemented for `Example`, which is required by `{float}: Into<_>`
   |
   = help: the following other types implement trait `From<T>`:
             <Example as From<()>>
             <Example as From<String>>
   = note: required for `{float}` to implement `Into<Example>`
